pulpcore/app/viewsets/exporter.py

```diff
@@ -71,3 +81,46 @@ class PulpExporterViewSet(ExporterViewSet):
     endpoint_name = 'pulp'
     serializer_class = PulpExporterSerializer
     queryset = PulpExporter.objects.all()
+
+
+class ExportViewSet(NamedModelViewSet,
+                    mixins.CreateModelMixin,
+                    mixins.RetrieveModelMixin,
+                    mixins.ListModelMixin,
+                    mixins.DestroyModelMixin):
+    """
+    ViewSet for viewing exports from an Exporter.
+    """
+    endpoint_name = 'exports'
+    nest_prefix = 'exporters'
+    router_lookup = 'export'
+    lookup_field = 'pk'
+    parent_lookup_kwargs = {'exporter_pk': 'exporter__pk'}
+    serializer_class = ExportSerializer
+    queryset = Export.objects.all()
+    parent_viewset = ExporterViewSet
+
+
+class PulpExportViewSet(ExportViewSet,
+                    mixins.CreateModelMixin,
+                    mixins.RetrieveModelMixin,
+                    mixins.ListModelMixin,
+                    mixins.DestroyModelMixin):
+    """
+    ViewSet for viewing exports from a PulpExporter.
+    """
+    parent_viewset = PulpExporterViewSet
+
+    def create(self, request, exporter_pk):
+        """
+        Export a set of repository-versions.
+
+        """
+        exporter = PulpExporter.objects.get(pk=exporter_pk).cast()
+
+        result = enqueue_with_reservation(
+            exporter.pulp_export,
```

Comment:
I know I created this pattern for file system exporter, but I think the code should probably just live in `/app/tasks/export.py`. And I would probably just drop this `pulp_export()` method.

Reply:
Refactored a number of things and repushed.